internal/pkg/provision/providers/docker/reflect.go

```diff
@@ -7,12 +7,13 @@ package docker
 import (
 	"context"
 	"net"
+	"strconv"
 
 	"github.com/talos-systems/talos/internal/pkg/provision"
 	"github.com/talos-systems/talos/pkg/config/machine"
 )
 
-func (p *provisioner) Reflect(ctx context.Context, clusterName string) (provision.Cluster, error) {
+func (p *provisioner) Reflect(ctx context.Context, clusterName, clustersDirectory string) (provision.Cluster, error) {
```

Comment:
I think we should refer to `clustersDirectory` as `stateDirectory`. It is a little bit more clear IMO (aligns with `--state` flag as well).